From ef8edc4005ff946df64ded758190244181972d95 Mon Sep 17 00:00:00 2001
From: xiaoyao <xiaoyao@rock-chips.com>
Date: Thu, 31 May 2018 09:47:37 +0800
Subject: [PATCH] net: wifi: cypress: fix some issues

  1: remove chipvendor id
  2: disable auto sleep
  3: add rockchip_wifi_get_oob_irq_flag

Change-Id: I28161c12ebd6e625815d09a0b670a923eda475d9
Signed-off-by: xiaoyao <xiaoyao@rock-chips.com>
---
 .../wireless/rockchip_wlan/cywdhd/bcmdhd/dhd_gpio.c   | 11 ++++++++++-
 .../wireless/rockchip_wlan/cywdhd/bcmdhd/dhd_linux.c  |  8 ++++----
 2 files changed, 14 insertions(+), 5 deletions(-)

diff --git a/drivers/net/wireless/rockchip_wlan/cywdhd/bcmdhd/dhd_gpio.c b/drivers/net/wireless/rockchip_wlan/cywdhd/bcmdhd/dhd_gpio.c
index 6ce071d29202..f7c1bbd66197 100644
--- a/drivers/net/wireless/rockchip_wlan/cywdhd/bcmdhd/dhd_gpio.c
+++ b/drivers/net/wireless/rockchip_wlan/cywdhd/bcmdhd/dhd_gpio.c
@@ -33,7 +33,7 @@ struct resource dhd_wlan_resources = {
 		.name	= "bcmdhd_wlan_irq",
 		.start	= 0,
 		.end	= 0,
-		.flags	= IORESOURCE_IRQ | IORESOURCE_IRQ_SHAREABLE	| IORESOURCE_IRQ_HIGHLEVEL,
+		.flags	= IORESOURCE_IRQ | IORESOURCE_IRQ_SHAREABLE,
 };
 
 static struct cntry_locales_custom brcm_wlan_translate_custom_table[] = {
@@ -105,6 +105,7 @@ static void *dhd_wlan_get_country_code(char *ccode
 int dhd_wlan_init_plat_data(void)
 {
     uint irq;
+	int irq_flags = -1;
 
     irq = rockchip_wifi_get_oob_irq();
 
@@ -113,5 +114,13 @@ int dhd_wlan_init_plat_data(void)
     dhd_wlan_resources.start = irq;
     dhd_wlan_resources.end = irq;
 
+	irq_flags = rockchip_wifi_get_oob_irq_flag();
+	if (irq_flags == 1)
+		dhd_wlan_resources.flags |= IORESOURCE_IRQ_HIGHLEVEL;
+	else if (irq_flags == 0)
+		dhd_wlan_resources.flags |= IORESOURCE_IRQ_LOWLEVEL;
+	else
+		pr_warn("%s: unknown oob irqflags !\n", __func__);
+
 	return 0;
 }
diff --git a/drivers/net/wireless/rockchip_wlan/cywdhd/bcmdhd/dhd_linux.c b/drivers/net/wireless/rockchip_wlan/cywdhd/bcmdhd/dhd_linux.c
index 6643a28fab90..ab17c018bb86 100644
--- a/drivers/net/wireless/rockchip_wlan/cywdhd/bcmdhd/dhd_linux.c
+++ b/drivers/net/wireless/rockchip_wlan/cywdhd/bcmdhd/dhd_linux.c
@@ -759,7 +759,8 @@ uint dhd_console_ms = 0;
 module_param(dhd_console_ms, uint, 0644);
 #endif /* defined(DHD_DEBUG) */
 
-uint dhd_slpauto = TRUE;
+//uint dhd_slpauto = TRUE;
+uint dhd_slpauto = FALSE;
 module_param(dhd_slpauto, uint, 0);
 
 #ifdef PKT_FILTER_SUPPORT
@@ -7205,14 +7206,13 @@ bool dhd_update_fw_nv_path(dhd_info_t *dhdinfo)
 	if (nvram_path[0] != '\0')
 		nv = nvram_path;
 
-	if (cis_device == BCM43455_CHIP_ID && cis_chipvendor == 0x81) {
+	if (cis_device == BCM43455_CHIP_ID) {
 		DHD_ERROR(("Adding 43455 firmware and NVRAM path by CIS\n"
 			"\tfirmware path: %s\n"
 			"\tNVRAM path:    %s\n", cis_fw_43455_path, cis_nv_43455_path));
 		fw = cis_fw_43455_path;
 		nv = cis_nv_43455_path;
-	}
-	else if (cis_device == BCM4354_CHIP_ID) {
+	} else if (cis_device == BCM4354_CHIP_ID) {
 		DHD_ERROR(("Adding 4354 firmware and NVRAM path by CIS\n"
 			"\tfirmware path: %s\n"
 			"\tNVRAM path:    %s\n", cis_fw_4354_path, cis_nv_4354_path));
-- 
2.35.3

